### PR TITLE
Switch to using Guice

### LINKED ISSUE #1: Simplify dependency injection with Guice (and remove static variables)
Remove statics and add DI via Guice.

### PR DESCRIPTION
**Pull request by bwillard**
Closes #1

### CODE REVIEW

File: portability-worker/src/main/resources/settings/common.yaml [hunk 1]
```diff
@@ -1,2 +1,10 @@
...
\ No newline at end of file
+cloud: LOCAL
+serviceProviderClasses: [
+  "org.dataportabilityproject.serviceProviders.flickr.FlickrServiceProvider",
+  "org.dataportabilityproject.serviceProviders.google.GoogleServiceProvider",
+  "org.dataportabilityproject.serviceProviders.instagram.InstagramServiceProvider",
+  "org.dataportabilityproject.serviceProviders.microsoft.MicrosoftServiceProvider",
+  "org.dataportabilityproject.serviceProviders.rememberTheMilk.RememberTheMilkServiceProvider",
+  "org.dataportabilityproject.serviceProviders.smugmug.SmugMugServiceProvider",
+]
```
Comment (holachuy) [inline portability-worker/src/main/resources/settings/common.yaml]: new line

File: portability-worker/src/main/resources/settings/common.yaml [hunk 2]
```diff
@@ -1,2 +1,10 @@
 env: LOCAL
-cloud: LOCAL
\ No newline at end of file
+cloud: LOCAL
+serviceProviderClasses: [
```
Comment (holachuy) [inline portability-worker/src/main/resources/settings/common.yaml]: Isn't this generated by the build script?

File: portability-core/src/main/java/org/dataportabilityproject/shared/AppCredentialFactory.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package org.dataportabilityproject.shared;
```
Comment (holachuy) [inline portability-core/src/main/java/org/dataportabilityproject/shared/AppCredentialFactory.java]: Add header

File: portability-core/src/main/java/org/dataportabilityproject/shared/CloudAppCredentialFactory.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
+package org.dataportabilityproject.shared;
```
Comment (holachuy) [inline portability-core/src/main/java/org/dataportabilityproject/shared/CloudAppCredentialFactory.java]: Add header

File: portability-core/src/main/java/org/dataportabilityproject/shared/local/LocalAppCredentialFactory.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
+package org.dataportabilityproject.shared.local;
```
Comment (holachuy) [inline portability-core/src/main/java/org/dataportabilityproject/shared/local/LocalAppCredentialFactory.java]: Add header

File: portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerImpl.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
+package org.dataportabilityproject.worker;
```
Comment (holachuy) [inline portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerImpl.java]: Add header

File: portability-core/src/main/java/org/dataportabilityproject/cloud/local/InMemoryPersistentKeyValueStore.java [hunk 7]
```diff
@@ -20,7 +20,7 @@
 import org.dataportabilityproject.cloud.interfaces.PersistentKeyValueStore;
 
 /** In-memory implementation of backend storage. */
-final class InMemoryPersistentKeyValueStore implements PersistentKeyValueStore {
+public final class InMemoryPersistentKeyValueStore implements PersistentKeyValueStore {
```
Comment (holachuy) [inline portability-core/src/main/java/org/dataportabilityproject/cloud/local/InMemoryPersistentKeyValueStore.java]: public or VisibleForTesting?

File: portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerImpl.java [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+    // Start the processing
+    PortabilityJob job = getJob(jobDao);
+
+    // Only load the two providers that are doing actually work.
+    // TODO(willard): Only load two needed services here, after converting service name to class
+    // name in the DAO.
+
+    processJob(job);
+    logger.info("Successfully processed jobId: {}", WorkerJobMetadata.getInstance().getJobId());
```
Comment (holachuy) [inline portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerImpl.java]: In a follow up, we can move WorkerJobMetadata to a @Singleton class

File: portability-api/src/main/java/org/dataportabilityproject/webapp/CopySetupHandler.java [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+* limitations under the License.
+*/
+package org.dataportabilityproject.webapp;
+
+import com.google.inject.Inject;
+import org.dataportabilityproject.ServiceProviderRegistry;
+import org.dataportabilityproject.job.JobDao;
+import org.dataportabilityproject.shared.settings.CommonSettings;
+
+final class CopySetupHandler extends SetupHandler {
```
Comment (seehamrun) [inline portability-api/src/main/java/org/dataportabilityproject/webapp/CopySetupHandler.java]: nit: comment top level class.

File: portability-api/src/main/java/org/dataportabilityproject/webapp/ApiServer.java [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+import com.sun.net.httpserver.HttpServer;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.Executors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class ApiServer {
```
Comment (seehamrun) [inline portability-api/src/main/java/org/dataportabilityproject/webapp/ApiServer.java]: Nit comment top level class

File: portability-api/src/main/java/org/dataportabilityproject/webapp/ImportSetupHandler.java [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+* limitations under the License.
+*/
+package org.dataportabilityproject.webapp;
+
+import com.google.inject.Inject;
+import org.dataportabilityproject.ServiceProviderRegistry;
+import org.dataportabilityproject.job.JobDao;
+import org.dataportabilityproject.shared.settings.CommonSettings;
+
+final class ImportSetupHandler extends SetupHandler {
```
Comment (seehamrun) [inline portability-api/src/main/java/org/dataportabilityproject/webapp/ImportSetupHandler.java]: nit: comment top level class

File: portability-api/src/main/java/org/dataportabilityproject/webapp/SetupHandler.java [hunk 12]
```diff
@@ -28,35 +28,41 @@
...
 import org.dataportabilityproject.job.JobUtils;
 import org.dataportabilityproject.job.PortabilityJob;
 import org.dataportabilityproject.shared.auth.AuthFlowInitiator;
 import org.dataportabilityproject.shared.auth.OnlineAuthDataGenerator;
+import org.dataportabilityproject.shared.settings.CommonSettings;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * HttpHandler for ImportSetup service
```
Comment (seehamrun) [inline portability-api/src/main/java/org/dataportabilityproject/webapp/SetupHandler.java]: Update comment on class

File: portability-core/src/main/java/org/dataportabilityproject/shared/AppCredentialFactory.java (hunk 3 above)
Comment (bwillard) [reply portability-core/src/main/java/org/dataportabilityproject/shared/AppCredentialFactory.java]: Done.

File: portability-core/src/main/java/org/dataportabilityproject/shared/CloudAppCredentialFactory.java (hunk 4 above)
Comment (bwillard) [reply portability-core/src/main/java/org/dataportabilityproject/shared/CloudAppCredentialFactory.java]: Done

File: portability-core/src/main/java/org/dataportabilityproject/shared/local/LocalAppCredentialFactory.java (hunk 5 above)
Comment (bwillard) [reply portability-core/src/main/java/org/dataportabilityproject/shared/local/LocalAppCredentialFactory.java]: Done

File: portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerImpl.java (hunk 6 above)
Comment (bwillard) [reply portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerImpl.java]: Done

File: portability-worker/src/main/resources/settings/common.yaml (hunk 2 above)
Comment (bwillard) [reply portability-worker/src/main/resources/settings/common.yaml]: It is removed.

File: portability-api/src/main/java/org/dataportabilityproject/webapp/ApiServer.java (hunk 10 above)
Comment (bwillard) [reply portability-api/src/main/java/org/dataportabilityproject/webapp/ApiServer.java]: Done

File: portability-api/src/main/java/org/dataportabilityproject/webapp/CopySetupHandler.java (hunk 9 above)
Comment (bwillard) [reply portability-api/src/main/java/org/dataportabilityproject/webapp/CopySetupHandler.java]: Done

File: portability-api/src/main/java/org/dataportabilityproject/webapp/ImportSetupHandler.java (hunk 11 above)
Comment (bwillard) [reply portability-api/src/main/java/org/dataportabilityproject/webapp/ImportSetupHandler.java]: Done

File: portability-api/src/main/java/org/dataportabilityproject/webapp/SetupHandler.java (hunk 12 above)
Comment (bwillard) [reply portability-api/src/main/java/org/dataportabilityproject/webapp/SetupHandler.java]: Done.

File: portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerImpl.java (hunk 8 above)
Comment (bwillard) [reply portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerImpl.java]: Ba Humbug, why wait till a new CL, I added that change to this one.

File: portability-worker/src/main/resources/settings/common.yaml (hunk 1 above)
Comment (bwillard) [reply portability-worker/src/main/resources/settings/common.yaml]: I just deleted the file

File: portability-core/src/main/java/org/dataportabilityproject/cloud/local/InMemoryPersistentKeyValueStore.java (hunk 7 above)
Comment (bwillard) [reply portability-core/src/main/java/org/dataportabilityproject/cloud/local/InMemoryPersistentKeyValueStore.java]: Public, its used across packages for testing.

**Comment by bwillard** [comment]
PTAL

**Pull request merged**